src/Symfony/Contracts/Deprecation/deprecated.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Triggers a deprecation.
+ *
+ * As recommended for prod, turn the "zend.assertions" ini setting to 0 or -1 to disable deprecation notices.
+ * Alternatively, provide your own implementation of the function and list "symfony/deprecation-contracts"
+ * in the "replace" section of your root composer.json if you need any custom behavior.
+ *
+ * The function doesn't use type hints nor variadics on purpose, to make it as fast as possibly achievable.
+ *
+ * @param string $package The name of the package that is triggering the deprecation
+ * @param string $version The version of the package that introduced the deprecation
+ * @param string $message The message of the deprecation; more arguments can be provided:
+ *                        they will be inserted in the message using printf() formatting
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+function deprecated($package, $version, $message)
+{
+    assert(@trigger_error(
+        ($package || $version ? "Since $package $version: " : '')
+        .(3 < func_num_args() ? vprintf($message, array_slice(func_get_args(), 3)) : $message),
```

Comment:
1. It's not possible to use variadic here ?
2. Is it intended to use `vprintf` and not `sprintf`, in the PR description, you talk about `sprintf`

Reply:
1. yes it is, but it wouldn't provide anything - it would even be a bit slower
2. description updated :)

Reply:
I did a [bench](https://github.com/lyrixx/php-bench/commit/a9bed4b5fbefdadcc1632487126f5d9eaf11ca26) about variadic

```
+------------------------------+---------+-------+
| subject                      | mean    | diff  |
+------------------------------+---------+-------+
| benchWithAndEmpty            | 0.062μs | 1.15x |
| benchWithAndEmptyNotEmpty    | 0.110μs | 2.05x |
| benchWithoutAndEmpty         | 0.053μs | 1.00x |
| benchWithoutAndEmptyNotEmpty | 0.113μs | 2.12x |
+------------------------------+---------+-------+
```

It's faster when "$args" is not empty :joy: 
I would prefer variadic version, perso it's easier to read

Reply:
Actually, the code is optimized for prod setup, ie when zend.assertions=0